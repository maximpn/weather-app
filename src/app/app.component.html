<mat-toolbar color="primary">
  <span>WeatherApp</span>
</mat-toolbar>

<router-outlet></router-outlet>
