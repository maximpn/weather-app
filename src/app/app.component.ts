import { Component } from '@angular/core';

@Component({
  selector: 'wa-root',
  templateUrl: 'app.component.html',
})
export class AppComponent {}
