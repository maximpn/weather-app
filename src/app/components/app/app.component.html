<mat-toolbar color="primary" class="mat-elevation-z6">
  <mat-icon>wb_sunny</mat-icon>
  <span routerLink="/" class="app-name">WeatherApp</span>
  <a href="https://github.com/mpalenov/weather-app" target="_blank" mat-button class="example-icon">
    <mat-icon svgIcon="github"></mat-icon>&nbsp;&nbsp;GitHub
  </a>
</mat-toolbar>
<mat-progress-bar *ngIf="loading$ | async" color="accent" mode="indeterminate"></mat-progress-bar>

<router-outlet></router-outlet>
