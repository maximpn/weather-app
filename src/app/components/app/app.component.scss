@import 'variables';

:host {
  display: flex;
  flex-direction: column;
  height: 100%;
}

::ng-deep router-outlet + * {
  margin-top: 64px;
  flex: 1;

  @media (max-width: $mat-toolbar-breakpoint) {
    margin-top: 56px;
  }
}

.app-name {
  margin-left: 10px;
  cursor: pointer;
}

.mat-toolbar {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
