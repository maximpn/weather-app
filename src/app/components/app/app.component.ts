import { Component } from '@angular/core';
import { NavigationEnd, NavigationStart, Router } from '@angular/router';
import { filter, map } from 'rxjs/operators';

@Component({
  selector: 'wa-root',
  templateUrl: 'app.component.html',
  styleUrls: ['app.component.scss'],
})
export class AppComponent {
  loading$ = this.router.events.pipe(
    filter(e => e instanceof NavigationStart ||  e instanceof NavigationEnd),
    map(e => e instanceof NavigationStart),
  );

  constructor(private readonly router: Router) {}
}
