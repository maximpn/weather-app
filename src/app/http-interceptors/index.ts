import { HTTP_INTERCEPTORS } from '@angular/common/http';

import { AccessTokenInterceptor } from './access-token.interceptor';
import { ApiEndpointInterceptor } from './api-endpoint.interceptor';
import { HttpErrorInterceptor } from './error.interceptor';

export const httpInterceptorProviders = [
  { provide: HTTP_INTERCEPTORS, useClass: AccessTokenInterceptor, multi: true },
  { provide: HTTP_INTERCEPTORS, useClass: ApiEndpointInterceptor, multi: true },
  { provide: HTTP_INTERCEPTORS, useClass: HttpErrorInterceptor, multi: true },
];
