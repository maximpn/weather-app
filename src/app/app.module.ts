import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { httpInterceptorProviders } from './http-interceptors';
import { API_ACCESS_TOKEN } from './services/api-access-token';
import { API_ENDPOINT } from './services/api-endpoint';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
  ],
  providers: [
    {
      provide: API_ACCESS_TOKEN,
      useValue: {
        paramName: 'APPID',
        value: 'dfcc4c8ffb4d85e5183930398cbcacff',
      },
    },
    {
      provide: API_ENDPOINT,
      useValue: {
        endpoint: 'http://api.openweathermap.org/data',
        version: '2.5',
      },
    },
    httpInterceptorProviders,
  ],
  bootstrap: [AppComponent],
})
export class AppModule { }
