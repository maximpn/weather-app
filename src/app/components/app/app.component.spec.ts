import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Event, NavigationStart, Router } from '@angular/router';
import { Subject } from 'rxjs';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;

  const routerEvents = new Subject<Event>();

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        AppComponent,
      ],
      providers: [
        { provide: Router, useValue: { events: routerEvents } },
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(AppComponent);

    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render the component with the loader', () => {
    routerEvents.next(new NavigationStart(1, 'some/url'));

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
