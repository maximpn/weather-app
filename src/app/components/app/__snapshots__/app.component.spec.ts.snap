// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render the component 1`] = `
<div
  id="root0"
>
  <mat-toolbar
    class="mat-elevation-z6"
    color="primary"
  >
    <mat-icon>
      wb_sunny
    </mat-icon>
    <span
      class="app-name"
      routerlink="/"
    >
      WeatherApp
    </span>
    <a
      class="example-icon"
      href="https://github.com/mpalenov/weather-app"
      mat-button=""
      target="_blank"
    >
      <mat-icon
        svgicon="github"
      />
        GitHub 
    </a>
  </mat-toolbar>
  
  <router-outlet />
</div>
`;

exports[`AppComponent should render the component with the loader 1`] = `
<div
  id="root1"
>
  <mat-toolbar
    class="mat-elevation-z6"
    color="primary"
  >
    <mat-icon>
      wb_sunny
    </mat-icon>
    <span
      class="app-name"
      routerlink="/"
    >
      WeatherApp
    </span>
    <a
      class="example-icon"
      href="https://github.com/mpalenov/weather-app"
      mat-button=""
      target="_blank"
    >
      <mat-icon
        svgicon="github"
      />
        GitHub 
    </a>
  </mat-toolbar>
  
  <mat-progress-bar
    color="accent"
    mode="indeterminate"
  />
  <router-outlet />
</div>
`;
